ui/hooks/snaps/useTransactionInsightSnaps.js

```diff
@@ -11,38 +11,26 @@ export function useTransactionInsightSnaps({
   chainId,
   origin,
   insightSnaps,
-  eagerFetching = true,
-  insightSnapId = '',
 }) {
   const subjects = useSelector(getPermissionSubjectsDeepEqual);
 
   const [loading, setLoading] = useState(true);
   const [data, setData] = useState(undefined);
-  ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
-  const [hasFetchedV2Insight, setHasFetchedV2Insight] = useState(false);
-  ///: END:ONLY_INCLUDE_IF
+  const [hasFetchedInsight, setHasFetchedInsight] = useState(false);
+
   useEffect(() => {
     let cancelled = false;
 
     async function fetchInsight() {
-      ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
-      if (hasFetchedV2Insight) {
-        setLoading(false);
-        return;
-      }
-      ///: END:ONLY_INCLUDE_IF
-
-      if (!eagerFetching) {
+      if (hasFetchedInsight) {
         setLoading(false);
```

Comment:
Do we still need this? The signature insights implementation does not have this?

Reply:
This was to prevent re-fetching iirc, tx params can change i.e. gas. For signatures the params are static, so this wasn't needed for signature insights.

Reply:
Oh that might be the case yeah 🤔